sdk/python/kfp/compiler/compiler_test.py

```diff
@@ -909,6 +909,70 @@ def my_pipeline() -> NamedTuple('Outputs', [
             ]):
                 task = print_and_return(text='Hello')
 
+    def test_pipeline_with_parameterized_container_image(self):
+        with tempfile.TemporaryDirectory() as tmpdir:
```

Comment:
If you use pytest, you can simply pass the `tmp_path` fixture as a parameter in your test function. Then, you won't need the context manager. Also `tmp_path` returns a pathlib.Path object by default so you don't need the conversion with `os.path.join()` later.

Reply:
TIL! unfortunately in this case KFP uses abseil testing framework, which is more like python's unittest framewrok, so this is not available to us 😢

Reply:
Ok thanks for letting me know! I got confused by `pipelines/pytest.ini`.